src/components/Organisms/InformationForm.js

```diff
@@ -1,250 +1,339 @@
-import React, { createRef, useRef } from 'react';
-import { Button, Form, TextInput, Select, SelectItem } from "carbon-components-react";
-import {createUseStyles} from 'react-jss';
-import stateList from "../../data/StateList";
-
+import React, { useState, } from 'react';
+import {
+  Button, Form, TextInput, Select, SelectItem,
+} from 'carbon-components-react';
+import { createUseStyles } from 'react-jss';
+import stateList from '../../data/StateList';
+import {emailPattern, addressPattern, namePattern, phoneNumberPattern, zipPattern, ssnPattern} from '../../data/Patterns';
+import apiCall from '../../classes/api/lumedicApi';
 const useStyle = createUseStyles({
-  informationForm:{
+  informationForm: {
     display: 'Flex',
     backgroundColor: 'white',
     flexWrap: 'wrap',
     marginTop: '10vw',
     marginBottom: '10vw',
     marginRight: '25vw',
     marginLeft: '25vw',
-    textAlign: 'left'
+    textAlign: 'left',
+  },
+  submitButton: {
+    display: 'flex',
+    backgroundColor: 'purple',
+    fontWeight: 'bold',
+    width: '15vw'
+  },
+  lableText: {
+    color: '#525252',
+    fontSize: '10pt',
+    whiteSpace: 'pre-wrap',
   },
-submitButton:{
-  backgroundColor: 'purple',
-  fontWeight: 'bold',
-  alignContent: 'center',
-  textAlign: 'center'
-},
-lableTextMobile:{
-  color: '#525252',
-  fontSize: '10pt',
-  whiteSpace: 'pre-wrap'
-},
-lableTextHeading:{
-  color: '#525252',
-  fontSize: '12pt',
-  fontWeight: 'bold'
-},
-lableTextPrivacy:{
-  color: '#525252',
-  fontSize: '10pt',
-  fontWeight: 'bold'
-},
-TextInputFields:{
+  lableTextHeading: {
+    color: '#525252',
+    fontSize: '12pt',
+    fontWeight: 'bold',
+  },
+  lableTextPrivacy: {
+    color: '#525252',
+    fontSize: '10pt',
+    fontWeight: 'bold',
+    display: 'flex'
+  },
+  TextInputFields: {
     paddingLeft: '2rem',
     paddingRight: '2rem',
-    width: '15vw'
-}
-})
+    width: '15vw',
+  },
+  NameBoxes: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    width: '20vw'
+  },
+  DOBSSN: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    width: '20vw'
+  },
+  ContactInfo: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    width: '20vw'
+  },
+  AddressInfo: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    width: '20vw'
+  }
+});
 
+const InformationForm = () => {
 
 
+  const [values, setValues] = useState({
+  firstName:"",
+  lastName:"",
+  dateOfBirth:"",
+  email:"",
+  last4SSN:"",
+  phoneNumber:"",
+  addressLine1:"",
+  addressLine2:"",
+  city:"",
+  state:"",
+  zip:"",
+  credName:"COVID-19 Vaccine"});
 
+  const [validity, setValidity] = useState({
+  firstName:true,
+  lastName:true,
+  dateOfBirth:true,
+  email:true,
+  last4SSN:true,
+  phoneNumber:true,
+  addressLine1:true,
+  addressLine2:true,
+  city:true,
+  state:true,
+  zip:true});
 
+  const [patternValidity, setPatternValidity] = useState({
```

Comment:
do we really require two different state objects for checking the validity? can you try to implement with only one?